src/server/controllers/model.controller.ts

```diff
@@ -1169,3 +1170,22 @@ export const getAssociatedResourcesCardDataHandler = async ({
   }
 };
 // #endregion
+export const getModelByHashesHandler = async ({ input }: { input: ModelByHashesInput }) => {
+  const { hashes } = input;
+
+  const modelsByHashes = await dbRead.$queryRaw<
+    { userId: number; modelId: number; hash: string }[]
+  >`
+      SELECT
+        m."userId",
+        m."id",
+        mfh."hash"
+      FROM "ModelFileHash" mfh
+      JOIN "ModelFile" mf ON mf."id" = mfh."fileId"
+      JOIN "ModelVersion" mv ON mv."id" = mf."modelVersionId"
+      JOIN "Model" m ON mv."modelId" = m.id
+      WHERE LOWER(mfh."hash") IN (${Prisma.join(hashes.map((h) => h.toLowerCase()))});
```

Comment:
Make sure that `hashes` has item in it, otherwise this raw query will fail with the empty array 😬 I know, classic prisma